bin/pre-push.sh

```diff
@@ -4,7 +4,7 @@ PROTECTED_BRANCH="master"
 REMOTE_REF=$(echo "$HUSKY_GIT_STDIN" | cut -d " " -f 3)
 
 if [ -n "$REMOTE_REF" ]; then
-	if [ -z "${REMOTE_REF##*$PROTECTED_BRANCH*}" ]; then
+	if [ -z "${REMOTE_REF##$PROTECTED_BRANCH}" ]; then
 		printf "%sYou're about to push to master, is that what you intended? [y/N]: %s" "$(tput setaf 3)" "$(tput sgr0)"
```

Comment:
I think we could make the error messaging clearer for GUI clients. I propose before the prompt we put something like this:

```bash
if [ "$TERM" == "dumb" ]; then
    echo "You are unable to push to master using a GUI client!"
    exit 1
fi
```

Reply:
Good suggestion, thanks. Outputting it to STDERR so that GUI clients will actually display it, e.g. PHPStorm (but also GitHub Desktop only shows STDERR output): 
![Screenshot 2020-07-14 at 12 56 14](https://user-images.githubusercontent.com/2207451/87418083-6e71a600-c5d1-11ea-9083-f9f072a64cb9.png)

Reply:
Should be fixed by https://github.com/woocommerce/woocommerce/pull/27028/commits/74c320575d40d83e202ccec89931ebf4ca79c761.